packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -47,22 +49,69 @@ class ImageCache {
     _maximumSize = value;
     if (maximumSize == 0) {
       _cache.clear();
+      _sizeBytes = 0;
     } else {
-      while (_cache.length > maximumSize)
-        _cache.remove(_cache.keys.first);
+      _checkCacheSize();
     }
   }
 
+  /// The current number of cached entries.
+  int get currentSize => _cache.length;
+
+  /// Maximum size of entries to store in the cache in bytes.
+  ///
+  /// Once more than this amount of bytes have been cached, the
+  /// least-recently-used entry is evicted until there are fewer than the
+  /// maximum bytes.
+  int get maximumSizeBytes => _maximumSize;
+  int _maximumSizeBytes = _kDefaultSizeBytes;
+
+  /// Changes the maximum cache bytes.
+  ///
+  /// If the new size is smaller than the current size in bytes, the
+  /// extraneous elements are evicted immediately. Setting this to zero and then
+  /// returning it to its original value will therefore immediately clear the
+  /// cache.
+  set maximumSizeBytes(int value) {
+    assert(value != null);
+    assert(value >= 0);
+    if (value == _maximumSizeBytes)
+      return;
+    _maximumSizeBytes = value;
+    if (_maximumSizeBytes == 0) {
+      _cache.clear();
+      _sizeBytes = 0;
+    } else {
+      _checkCacheSize();
+    }
+  }
+
+  int _sizeBytes = 0;
+  /// The current size of cached entries in bytes.
+  int get sizeBytes => _sizeBytes;
+
   /// Evicts all entries from the cache.
   ///
   /// This is useful if, for instance, the root asset bundle has been updated
   /// and therefore new images must be obtained.
-  // TODO(ianh): Provide a way to target individual images. This is currently non-trivial
-  // because by the time we get to the imageCache, the keys we're using are opaque.
   void clear() {
     _cache.clear();
   }
 
+  /// Evicts a single entry from the cache, returning true if successful.
+  ///
+  /// [key] is usually an image's corresponding [ImageProvider]. For example,
+  /// to evict a network image like below:
+  ///
+  ///     new Image.network(url);
```

Comment:
use triple backtick blocks with dart set as the syntax highlight language

Reply:
Fixed